File: 2023/01/solve.py
with open('input.txt') as f:
	lines = f.read()

print(lines)

word_pairs = {
    "one": 1,
    "two": 2,
    "three": 3,
    "four": 4,
    "five": 5,
    "six": 6,
    "seven": 7,
    "eight": 8,
    "nine": 9,
}

result = 0

for line in lines.strip().split("\n"):
    first = None
    last = None
    s = ""

    for char in line:
        digit = None
        if char.isdigit():
            digit = char
        else:
            s += char;

            for word, num in word_pairs.items():
                if s.endswith(word):
                    digit = str(num)
        if digit is not None:
            last = digit
            if first is None:
                first = digit

    result += int(first+last)
    print(first, last)

print(result)

File: 2023/01/solve.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"unicode"
)

func main() {
	f, err := os.Open("input.txt")
	var lines [][]rune
	var ans int

	if err != nil {
		log.Fatal(err)
	}

	defer f.Close()

	scanner := bufio.NewScanner(f)

	for scanner.Scan() {
		line := []rune(scanner.Text())
		lines = append(lines, line)
	}

	for _, line := range lines {
		fmt.Println(string(line))

		var first rune
		for _, char := range line {
			if unicode.IsDigit(char) {
				first = char
				break
			}
		}

		var last rune
		for _, char := range line {
			if unicode.IsDigit(char) {
				last = char
			}
		}
		fmt.Print(string(first))
		fmt.Print(string(last))
		fmt.Println("")

		sum := string(first) + string(last)
		num, error := strconv.Atoi(sum)

		if error != nil {
			fmt.Println("Error converting string to integer:", err)
			return
		}

		ans += num
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	fmt.Println("The answer is: ", ans)
}
